main.tf

```diff
@@ -13,10 +13,23 @@ terraform {
 }
 
 locals {
-  cluster_name = var.cluster_name != null ? var.cluster_name : terraform.workspace
+  cluster_name   = var.cluster_name != null ? var.cluster_name : terraform.workspace
+  resource_group = var.resource_group_name != null ? data.azurerm_resource_group.k8s[0] : azurerm_resource_group.k8s[0]
```

Comment:
This is actually correct, it's getting an actual `azurerm_resource_group` configuration object from the data source if we're using an existing, otherwise it's getting the object from the one we created.